src/sentry/static/sentry/app/actionCreators/sentryAppComponents.jsx

```diff
@@ -3,7 +3,7 @@ import SentryAppComponentsActions from 'app/actions/sentryAppComponentActions';
 export function fetchSentryAppComponents(api, orgSlug, projectId) {
   const componentsUri = `/organizations/${orgSlug}/sentry-app-components/?projectId=${projectId}`;
 
-  const promise = api.requestPromise(componentsUri);
-  promise.then(res => SentryAppComponentsActions.loadComponents(res));
-  return promise;
+  return api.requestPromise(componentsUri).then(res => {
+    SentryAppComponentsActions.loadComponents(res);
+  });
```

Comment:
This could change behavior in callers as the caller is now getting a chained promise. This anonymous function should return res to ensure the next `then()` is triggered correctly.